test/functional/tests/lazy_writes/test_flush_huge_dirty_data.py

```diff
@@ -0,0 +1,162 @@
+#
+# Copyright(c) 2020 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import pytest
+from datetime import timedelta
+
+from api.cas import casadm
+from api.cas.cache_config import CacheMode, CacheModeTrait, CleaningPolicy, SeqCutOffPolicy
+from api.cas.cli import stop_cmd
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeLowerThan, DiskTypeSet
+from test_tools.disk_utils import Filesystem
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import IoEngine, ReadWrite
+from test_tools.fs_utils import remove
+from test_utils.filesystem.file import File
+from test_utils.os_utils import sync, Udev
+from test_utils.size import Size, Unit
+
+file_size = Size(420, Unit.GiB)
+bs = Size(64, Unit.MebiByte)
+mnt_point = "/mnt/cas/"
+
+
+@pytest.mark.parametrizex("fs", Filesystem)
+@pytest.mark.parametrize("cache_mode", CacheMode.with_traits(CacheModeTrait.LazyWrites))
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_flush_over_400_gibibytes_with_fs(cache_mode, fs):
+    """
+        title: Test of the ability to flush huge amount of dirty data with filesystem.
+        description: |
+          Flush cache when amount of dirty data in cache with core with filesystem exceeds 400 GiB.
+        pass_criteria:
+          - Flushing completes successfully without any errors.
+    """
+    with TestRun.step("Prepare devices for cache and core."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([file_size * 1.05])
+        cache_part = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        Udev.disable()
```

Comment:
You have to check if cache device size is greater than file size and the same for core. As a core you can use raid volume.

Reply:
I think that the device used as core (normal disk or raid) should be set in config, but checking the size is a good idea.